src/main/java/uk/gov/justice/hmpps/datacompliance/services/referral/ReferralResolutionService.java

```diff
@@ -116,13 +116,16 @@ private boolean anyPending(final List<RetentionCheck> retentionChecks) {
 
     private Optional<RetentionCheckDataDuplicate> findPotentialFalsePositiveRetention(final List<RetentionCheck> retentionChecks) {
 
-        final var dataDuplicateRetentions = retentionChecks.stream()
+        final var checksCausingRetention = retentionChecks.stream()
                 .filter(check -> check.isStatus(RETENTION_REQUIRED))
+                .collect(toList());
+
+        final var dataDuplicateRetentions = checksCausingRetention.stream()
                 .filter(check -> check.isType(DATA_DUPLICATE_DB) || check.isType(DATA_DUPLICATE_AP))
                 .map(RetentionCheckDataDuplicate.class::cast)
                 .collect(toList());
 
-        return dataDuplicateRetentions.size() == 1 ? dataDuplicateRetentions.stream().findFirst() : Optional.empty();
+        return checksCausingRetention.size() == 1 ? dataDuplicateRetentions.stream().findFirst() : Optional.empty();
```

Comment:
Correcting logic - want to count all checks causing retention, not just the data duplicate checks.